c2corg_ui/static/js/feed.js

```diff
@@ -114,16 +163,165 @@ app.FeedController = function(appAuthentication, appApi, appLang, imageUrl, ngeo
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.showAssoBlock = /** @type {boolean} */ (JSON.parse(window.localStorage.getItem('showAssoBlock') || 'true'));
+
+  /**
    * @type {ngeo.Location}
    * @public
    */
   this.ngeoLocation = ngeoLocation;
 
   this.getDocumentsFromFeed();
+  this.getLatestTopics_();
+  this.feedColumnManager();
+  this.getAnnouncement_();
 };
 
 
 /**
+ * init array for the column Manager
+ * @private
+ */
+app.FeedController.prototype.getAnnouncement_ = function() {
+
+  this.hasAnnounce = true;
+  this.announce = {'title':'','message': ''};
+
+};
+
+/**
+ * toggle block above forum topics list
+ @param {number} id
+ * @export
+ */
+app.FeedController.prototype.toggleBlock = function(id) {
+  switch (id) {
+    case 0:
+      this.showAssoBlock = !this.showAssoBlock;
+      window.localStorage.setItem('showAssoBlock', JSON.stringify(this.showAssoBlock));
+      break;
+    case 1:
+      this.showMobileBlock = !this.showMobileBlock;
+      window.localStorage.setItem('showMobileBlock', JSON.stringify(this.showMobileBlock));
+      break;
+    default:
+      break;
+  }
+};
+
+/**
+ * init array for the column Manager
+ * @private
+ */
+app.FeedController.prototype.initDocumentsCol_ = function() {
+  if (this.documentsCol[0] === null || this.documentsCol[0] == undefined) {
+    this.documentsCol[0] = Array();
+  }
+  if (this.documentsCol[1] === null || this.documentsCol[1] == undefined) {
+    this.documentsCol[1] = Array();
+  }
+  if (this.documentsCol[2] === null || this.documentsCol[2] == undefined) {
+    this.documentsCol[2] = Array();
+  }
+};
+
+/**
+ * refresh the feed column according the width
+ * @export
+ */
+app.FeedController.prototype.feedColumnManager = function() {
+
+  $(window).resize(function() {
+
+    if (window.innerWidth < 1400) {
+      if (this.nbCols_ != 1) {
+        this.documentsCol = Array();
+        this.documentsCol[0] = this.documents;
+        this.documentsCol[1] = Array();
+        this.documentsCol[2] = Array();
+
+        this.nbCols_ = 1;
+        this.scope_.$apply();
+      }
+
+    } else if (window.innerWidth >= 1400 && window.innerWidth < 2000) {
+
+
+      if (this.nbCols_ != 2) {
+        this.documentsCol = Array();
+        this.documentsCol[0] = Array();
+        this.documentsCol[1] = Array();
+        this.documentsCol[2] = Array();
+        this.nbCols_ = 2;
+
+        var height1_c2 = 0;
+        var height2_c2 = 0;
+
+        for (var i = 0,n = this.documents.length; i < n; i++) {
+
+          if (height1_c2 <= height2_c2) {
+            this.documentsCol[0].push(this.documents[i]);
+            height1_c2 = height1_c2 + this.sizeEstimator(this.documents[i]);
+          } else if (height2_c2 <= height1_c2) {
+            this.documentsCol[1].push(this.documents[i]);
+            height2_c2 = height2_c2 + this.sizeEstimator(this.documents[i]);
+          } else {
```

Comment:
Is this last block really needed? I mean it's a duplicate of the first block and will anyway never be executed since one of the 2 previous conditions is always true.

Then perhaps you could write
```
if (height1_c2 <= height2_c2) {
  this.documentsCol[0].push(this.documents[i]);
  height1_c2 += this.sizeEstimator(this.documents[i]);
} else {
  this.documentsCol[1].push(this.documents[i]);
  height2_c2 += this.sizeEstimator(this.documents[i]);
}
```